modules/run-vault/run-vault

```diff
@@ -103,14 +110,33 @@ function generate_vault_config {
   local readonly cluster_port="$4"
   local readonly config_dir="$5"
   local readonly user="$6"
+  local readonly enable_s3_backend="$7"
+  local readonly s3_bucket="$8"
+  local readonly s3_bucket_region="$9"
   local readonly config_path="$config_dir/$VAULT_CONFIG_FILE"
 
   local instance_ip_address
   instance_ip_address=$(get_instance_ip_address)
 
   log_info "Creating default Vault config file in $config_path"
-  cat > "$config_path" <<EOF
-storage "consul" {
+  cat >> "$config_path" <<EOF
```

Comment:
It's a bit hard to visualize this config file when it's broken up across several steps and if/else statement. Also, the Consul storage config ends up being copy/pasted twice. 

Perhaps a cleaner approach is to:

1. Define an `s3_config` variable that is set to empty string by default, but if `enable_s3_backend` is `true`, you set it to the `storage "s3" { ... }` block.

1. Define a `consul_storage_type` variable that is set to `storage` by default, but gets reset to `ha_storage` if `enable_s3_backend` is `true`. You can then define the Consul configuration exactly once in the format: `$consul_storage_type "consul" { ... }`.